Cabal/tests/UnitTests/Distribution/SPDX.hs

```diff
@@ -55,6 +57,63 @@ reassoc (EAnd a b) = case reassoc a of
     x       -> EAnd x (reassoc b)
 reassoc l = l
 
+-------------------------------------------------------------------------------
+-- isAcceptableLicence
+-------------------------------------------------------------------------------
+
+shouldAccept :: [License]
+shouldAccept = map License
+    [ simpleLicenseExpression GPL_2_0_only
+    , simpleLicenseExpression GPL_2_0_or_later
+    , simpleLicenseExpression BSD_2_Clause
+    , simpleLicenseExpression BSD_3_Clause
+    , simpleLicenseExpression MIT
+    , simpleLicenseExpression ISC
+    , simpleLicenseExpression MPL_2_0
+    , simpleLicenseExpression Apache_2_0
+    , simpleLicenseExpression CC0_1_0
+    , simpleLicenseExpression BSD_4_Clause `EOr` simpleLicenseExpression MIT
+    ]
+
+shouldReject :: [License]
+shouldReject = map License
+    [ simpleLicenseExpression BSD_4_Clause
+    , simpleLicenseExpression BSD_4_Clause `EAnd` simpleLicenseExpression MIT
+    ]
+
+-- | A sketch of what Hackage could accept
+--
+-- * NONE is rejected
+--
+-- * "or later" syntax (+ postfix) is rejected
+--
+-- * "WITH exc" exceptions are rejected
```

Comment:
Won't this reject [HaXml](http://hackage.haskell.org/package/HaXml-1.25.4/src/COPYRIGHT)?

Reply:
@ivan-m what would be the SPDX expression for accurately describing HaXml's licence?

Reply:
this thread ends up  listing more standard licenses that seem to take the place of LGPL+linking exception: https://softwareengineering.stackexchange.com/questions/179084/is-there-a-modified-lgpl-license-that-allows-static-linking